src/components/RegistrationForm.js

```diff
@@ -13,20 +15,35 @@ class RegistationForm extends Component {
 
   submitHandler = e => {
     e.preventDefault();
-    alert("I don't work yet");
+    const type = this.props.match.url.substring(1);
```

Comment:
If you combine the `login` and `signup` in your actions into a single `authenticate` function, you could lose a lot of this logic:

```suggestion
    const type = this.props.match.url.substring(1);
    this.props.authenticate(this.state, type);
```